GEODE-3269: Refactoring ShellCommands

PurelyApplied opened the pull request:
This PR reconciles #709 against the current state of `develop`, with minor additional refactorings.

Precheckin running.

-----

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

PurelyApplied commented:
Hold off on pulling this in.  There are issues in precheckin.

pdxrunner commented on geode-core/src/main/java/org/apache/geode/management/internal/web/controllers/ShellCommandsController.java:
```diff
@@ -97,7 +108,7 @@
     HttpHeaders respHeaders = new HttpHeaders();
     InputStreamResource isr;// if the command is successful, the output is the filepath,
-    // else we need to send the orignal result back so that the receiver will know to turn it
+    // else we need to send the original result back so that the receiver will know to turn it
```
The comment formatting could be made clearer. It's awkward to read jammed into the preceding declaration with no spacing.

PurelyApplied replied:
I agree!  I try to adjust them when I see things like this, but this one must've flown under my radar.

PurelyApplied commented:
@pdxrunner Do you have the second commit (`888f3a21d516a06d49e31930703475ca5f222da9`)?  It should have been removed along with some few other minor refactorings.

PurelyApplied commented:
Precheckin ran green-ish: one unrelated protobuf failure which should not be resolved, and travis failure due to bad rebasing in the GfshRule migration to `geode-junit`.

Squashed my two commits and rebased to develop, which has shifted slightly but should not be affected by this commit.

pdxrunner commented:
I'll pull this after Travis finishes

jinmeiliao commented on geode-core/src/main/java/org/apache/geode/management/internal/web/controllers/ShellCommandsController.java:
```diff
@@ -53,7 +53,18 @@
  * 
- * @see org.apache.geode.management.internal.cli.commands.ShellCommands
+ * @see org.apache.geode.management.internal.cli.commands.ConnectCommand
+ * @see org.apache.geode.management.internal.cli.commands.DebugCommand
```
I think we should get rid of all those @see. They are not even related to this controller anymore.

PurelyApplied commented:
It looks like I accidentally deleted the remote branch.  Restored and reopened, with the extraneous javadoc `@see`s removed.
